frontend/src/core/editor/selectors.js

```diff
@@ -60,9 +62,25 @@ export function _existingTranslation(
         }
         // If translation is a string, from the generic editor.
         else {
-            existingTranslation = history.translations.find(
-                t => t.string === translation
-            )
+            // Except it might actually be a simple Fluent message from the SimpleEditor.
+            if (entity.format === 'ftl') {
+                // This case happens when the format is Fluent and the string is simple.
+                // We thus store a simplified version of the Fluent message in memory,
+                // but the history has actual Fluent-formatted strings (with ID).
+                // Thus, we need to reconstruct the Fluent message in order to be able
+                // to compare it with items in history.
+                const reconstructed = fluent.serializer.serializeEntry(
+                    fluent.getReconstructedMessage(entity.original, translation)
+                );
+                existingTranslation = history.translations.find(
+                    t => t.string === reconstructed
+                )
```

Comment:
You can move this block out of the `if-else` block and if you call `reconstructed` and `translation` by the same name, you don't need to repeat it below.

Reply:
As we talked about yesterday, this is very low value, and I actually like that we have different variable names in there, to make the difference more explicit. Hence, I'm not going to do it. :angel: